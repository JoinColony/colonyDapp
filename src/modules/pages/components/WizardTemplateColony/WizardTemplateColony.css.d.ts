export const query700: string;
export const layoutMain: string;
export const header: string;
export const steps: string;
export const content: string;
export const wallet: string;
export const address: string;
export const headerWallet: string;
export const noMoney: string;
export const yeihMoney: string;
export const moneyContainer: string;
export const hello: string;
export const copy: string;
