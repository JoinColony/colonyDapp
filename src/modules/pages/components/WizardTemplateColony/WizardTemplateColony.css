.layoutMain {
  composes: stretchVertical stretchHorizontal from '~styles/layout.css';
  overflow: auto;
  background-color: var(--grey-blue-2);
}

.header {
  display: flex;
  justify-content: space-between;
  padding-top: 8px;
  height: 60px;
  width: 100%;
}

.steps {
  flex: 1;
  margin-top: 5px;
}

.backButton {
  margin-left: 18px;
}

.content {
  composes: flexContainerColumn flexAlignCenter flexJustifyCenter from '~styles/layout.css';
  padding-bottom: 20px;
}

.wallet {
  display: flex;
  flex-direction: column;
  margin: 10px 19px 0px 0px;
}

.address {
  display: flex;
  justify-content: flex-end;
  justify-content: space-between;
  width: 150px;
}

.headerWallet {
  display: flex;
  margin: 7px 14px 0px 0px;
}

.balance {
  display: flex;
  justify-content: flex-end;
  color: var(--danger);
}

.balance::after {
  display: inline-block;
  margin-bottom: 1px;
  margin-left: 3px;
  height: 8px;
  width: 8px;
  position: relative;
  top: 5px;
  right: 74px;
  border-radius: 50%;
  background-color: var(--danger);
  content: '';
}
