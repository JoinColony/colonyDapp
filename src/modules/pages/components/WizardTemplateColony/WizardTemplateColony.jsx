/* @flow */
import type { Node } from 'react';

import React from 'react';

import { StepBar } from '~core/ProgressBar';
import { HistoryNavigation } from '~pages/NavigationWrapper';

import styles from './WizardTemplateColony.css';

type Props = {|
  children: Node,
  step?: number,
  stepCount?: number,
|};

const displayName = 'pages.WizardTemplateColony';

const WizardTemplateColony = ({ children, step, stepCount }: Props) => (
  <main className={styles.layoutMain}>
    <header className={styles.header}>
      <div className={styles.backButton}>
        <HistoryNavigation noText />
      </div>
      {stepCount && step && (
        <div className={styles.steps}>
          <StepBar step={step} stepCount={stepCount} />
        </div>
      )}
    </header>
    <article className={styles.content}>{children}</article>
  </main>
);

WizardTemplateColony.displayName = displayName;

export default WizardTemplateColony;
