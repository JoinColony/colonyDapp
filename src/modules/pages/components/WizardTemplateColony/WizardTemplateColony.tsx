import React, { ReactNode, useCallback } from 'react';
import { defineMessages, FormattedMessage } from 'react-intl';
import { parseEther } from 'ethers/utils';
import { Redirect } from 'react-router-dom';

import Numeral from '~core/Numeral';
import QRCode from '~core/QRCode';
import CopyableAddress from '~core/CopyableAddress';
import { HistoryNavigation } from '~pages/RouteLayouts';
import { useLoggedInUser } from '~data/index';
import { DEFAULT_NETWORK_TOKEN, ALLOWED_NETWORKS } from '~constants';
import { DASHBOARD_ROUTE } from '~routes/index';

import styles from './WizardTemplateColony.css';

export interface Props {
  children: ReactNode;
  previousStep: () => boolean;
  hideQR: boolean;
}

const MSG = defineMessages({
  wallet: {
    id: 'dashboard.CreateColonyWizard.StepUserENSName.wallet',
    defaultMessage: 'Hello ',
  },
});

const displayName = 'pages.WizardTemplateColony';

const WizardTemplateColony = ({
  children,
  previousStep,
  hideQR = false,
}: Props) => {
  const { balance, walletAddress, networkId } = useLoggedInUser();
  const customHandler = useCallback(() => previousStep(), [previousStep]);
  const ethBalance = parseEther(balance);
  const isNetworkAllowed = !!ALLOWED_NETWORKS[networkId || 1];

  if (!isNetworkAllowed) {
    return <Redirect to={DASHBOARD_ROUTE} />;
  }

  return (
    <main className={styles.layoutMain}>
      <header className={styles.header}>
        <HistoryNavigation customHandler={customHandler} backText=" " />
        <div className={styles.headerWallet}>
          <div className={styles.wallet}>
            <div className={styles.address}>
              <span className={styles.hello}>
                <FormattedMessage {...MSG.wallet} />
              </span>
              <span className={styles.copy}>
                <CopyableAddress>{walletAddress}</CopyableAddress>
              </span>
            </div>
            <div className={styles.moneyContainer}>
              {ethBalance.isZero() ? (
                <div className={styles.noMoney}>
                  <Numeral
                    suffix={` ${DEFAULT_NETWORK_TOKEN.symbol}`}
                    unit="ether"
                    value={ethBalance}
                  />
                </div>
              ) : (
                <div className={styles.yeihMoney}>
                  <Numeral
                    suffix={` ${DEFAULT_NETWORK_TOKEN.symbol}`}
                    unit="ether"
                    value={ethBalance}
                  />
                </div>
              )}
            </div>
          </div>
          {!hideQR && <QRCode address={walletAddress} width={60} />}
        </div>
      </header>
      <article className={styles.content}>{children}</article>
    </main>
  );
};

WizardTemplateColony.displayName = displayName;

export default WizardTemplateColony;
