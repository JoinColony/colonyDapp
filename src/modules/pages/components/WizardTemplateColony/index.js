/* @flow */

export { default } from './WizardTemplateColony.jsx';
