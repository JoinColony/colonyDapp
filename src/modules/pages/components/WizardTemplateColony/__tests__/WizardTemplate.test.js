/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from 'testutils';

import WizardTemplateColony from '../WizardTemplateColony.jsx';

describe('WizardTemplateColony component', () => {
  test('Renders initial component', () => {
    const component = shallowWithIntl(
      <WizardTemplateColony stepCount={2} step={1} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
