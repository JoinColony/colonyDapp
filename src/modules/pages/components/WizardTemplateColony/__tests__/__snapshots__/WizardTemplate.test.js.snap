// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardTemplateColony component Renders initial component 1`] = `
<main>
  <header>
    <div>
      <withRouter(InjectIntl(pages.NavigationWrapper.HistoryNavigation))
        noText={true}
      />
    </div>
    <div>
      <InjectIntl(StepBar)
        step={1}
        stepCount={2}
      />
    </div>
  </header>
  <article />
</main>
`;
